app/models/user.rb

```diff
@@ -395,41 +395,45 @@ def external_usergroups
   end
 
   def self.try_to_auto_create_user(login, password)
-    return nil if login.blank? || password.blank?
-
-    # user is not yet registered, try to authenticate with available sources
+    return if login.blank? || password.blank?
+    # User is not yet registered, try to authenticate with available sources
     logger.debug "Attempting to log into an auth source as #{login} for account auto-creation"
-    if (attrs = AuthSource.authenticate(login, password))
-      attrs.delete(:dn)
-      user = new(attrs)
-
-      # If an invalid data returned from an authentication source for any attribute(s) then set its value as nil
-      saved_attrs = {}
-      unless user.valid?
-        user.errors.each do |attr|
-          saved_attrs[attr] = user[attr]
-          user[attr] = nil
-        end
+    return unless (attrs = AuthSource.authenticate(login, password))
+
+    attrs.delete(:dn)
+    user = new(attrs)
+    # Inherit taxonomies from authentication source on creation
```

Comment:
These four lines are the only thing that changed. The rest of the diff is shown as 'diff' because I changed the huge 'if' statement for an early return.